### ISSUE TITLE
Preserve extended attributes while copy

### ISSUE THREAD

**Issue by mc-butler** [MidnightCommander/mc#2468]
> [!IMPORTANT]
> This issue was migrated from Trac:
>
> | **Origin** | https://midnight-commander.org/ticket/2468 |
> | :------------- | :------------------------------------------------------------------------------------------------ |
> | **Reporter** | `Kroz` (kroz.nn@….com) |
> | **Mentions** | `gotar@….pl`, `sendsbeak@….com`, `andrey.gursky@…-mail.ua` |
> | **Keywords** | `extended`, `attributes` |

<p>
Extended attributes are nice to store comments and keywords to files.<br />
</p>
<p>
Set an extended attribute to a file:<br />
$ attr -s variable -V "value" filename.txt<br />
</p>
<p>
Now see it:<br />
$ attr -l filename.txt<br />
</p>
<p>
If you copy such file with mc, extended attribute will be absent for file's copy.<br />
It would be nice to have an option like "preserve Extended attributes" to let mc copy extended attributes as well.<br />
</p>

**Comment by shockme** [comment]
> Can you mention any managers supporting it by default? Maybe we can steal code from them.

Double Commander

**Comment by objectionoverruled** [comment]
Hi. Is there any news on this?
I am concerned about 2 topics:

1. F5 copy does not preserve XAttr. "Preserve attributes" does not cover this. // I have created a user menu command for this, but still
2. I can't do anything to show XAttr as a column in file listing. I would love to have "Tags" (user.xdg.tags XAttr) in file listing panel. Or, at least, in Info panel.
3. Being able to sort by tags would be the next bright future step... // I think I can do this with external panelize but... I'm not sure if I can use external panelize as simple as traversing the file system and having the default sort order based on tags... Never used it before.

If there are impassable obstacles, or complexities for implementing XAttr support in MC, can someone be so kind to briefly explain why?
Thank you.

**Comment by zyv** [comment]
> If there are impassable obstacles, or complexities for implementing XAttr support in MC, can someone be so kind to briefly explain why? Thank you.

I believe that I've already explained to you what the impassable obstacle at the moment is: lack of manpower and lack of interest. Is there anything unclear about this? If you were to provide a high-quality implementation with tests, we'd try to get it in. Otherwise, I'm afraid, it'll have to wait until better days.

P.S. Oh, and please don't just throw some "AI" tool on it and submit the results. This would be unhelpful; I could have done as much myself.

**Comment by objectionoverruled** [comment]
You're right, I forgot that I've already asked this question. Thanks for the reminder.
